### PR TITLE
MIAUI draft

### PR DESCRIPTION
**Pull request by gal42**
MIAUI sample draft for reference. Started to work on adding new task. Has bugs with InitializeComponent method

### CODE REVIEW

**Comment by officialgio** [comment]
Try deleting the old code regarding MVVM and INotifyProperyChanged - try using CommunityToolkit.Mvvm Nuget Package.

File: MIAUI/Views/NewTaskPage.xaml.cs [hunk 1]
```diff
@@ -5,9 +5,9 @@ namespace MIAUI.Views;
 public partial class NewTaskPage : ContentPage
 {
 	public Model.Task Task;
```
Comment (tj-devel709) [inline MIAUI/Views/NewTaskPage.xaml.cs:7]: Is this Task Variable is being used here?

**Review by rachelkang — CHANGES_REQUESTED** [review]
Awesome work so far! 

Let's just do a little bit of clean up before we merge this - delete all the code that isn't being used (usings, commented code, unused variables). Then, we should be good to move forward :)

File: MIAUI/ViewModels/BaseViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using CommunityToolkit.Mvvm.ComponentModel;
+using System.Collections.ObjectModel;
+using CommunityToolkit.Mvvm.Input;
+using System.Threading.Tasks;
+using MIAUI.Views;
+using Task = System.Threading.Tasks.Task;
+
+namespace MIAUI.ViewModels
+{
+    public partial class BaseViewModel : ObservableObject
```
Comment (davidortinau) [inline MIAUI/ViewModels/BaseViewModel.cs:21]: Nuke this file since you don't need it

File: MIAUI/ViewModels/SubtasksViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace MIAUI.ViewModels
+{
+    //[QueryProperty("Text", "Text")]
+    public partial class SubtasksViewModel : BaseViewModel
```
Comment (davidortinau) [inline MIAUI/ViewModels/SubtasksViewModel.cs:13]:
```suggestion
    [INotifyPropertyChanged]
    public partial class SubtasksViewModel
```

No longer need this

File: MIAUI/ViewModels/TasksViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+using CommunityToolkit.Mvvm.ComponentModel;
+using System.Collections.ObjectModel;
+using CommunityToolkit.Mvvm.Input;
+using System.Threading.Tasks;
+using MIAUI.Views;
+using Task = System.Threading.Tasks.Task;
+
+namespace MIAUI.ViewModels
+{
+    public partial class TasksViewModel : BaseViewModel
```
Comment (davidortinau) [inline MIAUI/ViewModels/TasksViewModel.cs:21]:
```suggestion
    [INotifyPropertyChanged]
    public partial class TasksViewModel
```

File: MIAUI/ViewModels/SubtasksViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
```
Comment (davidortinau) [inline MIAUI/ViewModels/SubtasksViewModel.cs:1]: How about adding a `GlobalUsings.cs` file to the root and moving the common using statements there to clean up your other files. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive#global-modifier

File: MIAUI/ViewModels/TasksViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        [ObservableProperty]
+        ObservableCollection<string> items;
+
+        [ObservableProperty]
+        bool isBusy = false;
+
+        [ObservableProperty]
+        string taskName; //Task's title
+
+        [ICommand]
```
Comment (davidortinau) [inline MIAUI/ViewModels/TasksViewModel.cs:37]: I happen to notice in the latest pre-release of the CommunityToolkit.Mvvm that ICommand is gone and everything seems to be RelayCommand. Probably no reason to adopt a pre-release, but just a heads up.

**Review by davidortinau — COMMENTED** [review]
Looking good!

As time permits you should refactor colors and control styles into static resources. https://docs.microsoft.com/en-us/dotnet/maui/user-interface/styles/xaml

**Pull request merged**